app/src/main/java/org/sagebionetworks/research/mpower/tracking/TrackingTabFragment.java

```diff
@@ -1,139 +1,139 @@
+
+    @BindView(R.id.tracking_status_bar)
+    TrackingStatusBar trackingStatusBar;
+
+    @BindView(R.id.togglebutton_loading)
+    ToggleButton scheduledActivitiesLoadingToggleButton;
+
+    @BindView(R.id.textview_scheduled_activities)
+    TextView scheduledActivitiesTextView;
+
+    @Inject
+    TrackingViewModelFactory trackingViewModelFactory;
+
+    private TrackingViewModel trackingViewModel;
+
+    private Unbinder unbinder;
+
+    public static TrackingTabFragment newInstance() {
+        return new TrackingTabFragment();
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        AndroidSupportInjection.inject(this);
+        super.onAttach(context);
+    }
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+            @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.tracking_fragment, container, false);
+        unbinder = ButterKnife.bind(this, view);
+        // Move the status bar down by the window insets.
+        OnApplyWindowInsetsListener listener = SystemWindowHelper.getOnApplyWindowInsetsListener(Direction.TOP);
+        ViewCompat.setOnApplyWindowInsetsListener(this.trackingStatusBar, listener);
+        return view;
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        ViewCompat.requestApplyInsets(this.getView());
+    }
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        trackingViewModel = ViewModelProviders.of(
+                this,
+                trackingViewModelFactory.create()).get(TrackingViewModel.class);
+
+        trackingViewModel.getScheduledActivitiesLiveData()
+                .observe(this, this::updateScheduledActivities);
+        trackingViewModel.getScheduledActivitiesLoadingErrorMessageLiveData()
+                .observe(this, this::updateScheduledActivitiesErrorMessage);
+        trackingViewModel.getScheduledActivitiesLoadingLiveData()
+                .observe(this, this::updateScheduledActivitiesLoading);
+    }
+
+    @Override
+    public void onDestroyView() {
+        super.onDestroyView();
+        unbinder.unbind();
+    }
+
+    @OnClick(R.id.button_reload)
+    void onReloadClicked() {
+        trackingViewModel.reload();
+    }
+
+    // these methods marked with @VisibleForTesting would be the ones to UI test. Since this is a dumb/passive view,
+    // the only work it should be doing is subscribing to POJOs and rendering them.
+    @VisibleForTesting
+    void updateScheduledActivities(List<ScheduledActivityView> scheduledActivityViews) {
+        StringBuilder sb = new StringBuilder();
+        for (ScheduledActivityView v : scheduledActivityViews) {
+            sb.append(v.scheduledActivityGuid + " ");
+        }
+    }
+
+    @VisibleForTesting
+    void updateScheduledActivitiesErrorMessage(String errorMessage) {
```

Comment:
Because this method will be present in every single fragment, I would suggest creating a Kotlin interface and base class for handling error messages and using class delegation to get the functionality for free in every Fragment without having to deal with a BridgeBaseFragment